healthlytix/slicer-plugins
# Write module for comparing multiple segs, one ROI at a time (Feature/rg 1659)

## Description

I've added yet another plugin for Slicer: Compare Segmentations. This is quite similar to the Seg Reviewer, except that it displays multiple segmentations, one ROI at a time. I've added a radio button to select the ROI; this button toggles the visibility of the ROIs across all of the labelers' segmentations.

## Data

The data is expected to be organized like so:

```
segmentations
├── labeler1
│   ├── case1
│   │   ├── im1.nii
│   │   ├── im2.nii
│   │   └── seg.nii
...
│   ├── case2
│   │   ├── im1.nii
│   │   ├── im2.nii
│   │   └── seg.nii
...
├── labeler2
│   ├── case1
│   │   ├── im1.nii
│   │   ├── im2.nii
│   │   └── seg.nii
...
│   ├── case2
│   │   ├── im1.nii
│   │   ├── im2.nii
│   │   └── seg.nii
...
```

Although it hasn't been tested too thoroughly, I tried to write the data loading function to handle missing data robustly. So, for example, for a given case, it will look in all of the labelers' folders for the MRIs; only one labeler needs to have the images. If a particular labeler is missing a seg on a certain case, it will simply load the other labelers' segs. If all segs are missing for a case, but the MRIs are present, it will just load the images. 

## Testing
* Download the `keating`, `vidic` and `smith` folders from [the dry run labeler folder](https://drive.google.com/drive/u/0/folders/1H8nZ1WGanzPkzhIHeyTjBWcbTbFvOddL). \
* Reorganize Robert's data by creating case folders for `smith` and put the seg files in the case folders. All three labeler folders should contain the same case folders.
* Install the module as usual
* Click on the "Select Data" button and select the  `keating`, `vidic` and `smith` folders
* Play around with it and make sure the widgets work as expected. Let me know if there are any issues or if there's any way to improve the visualizations.

## Issues
* Right now there are only 3 labeler colors; for more than 3 segs it will cycle the colors.
* I added a mostly-transparent fill because the outlines/contours alone thin and hard to see. I suspect the 2d fill will look worse as you add more segs
